tests/perf/benchmark.py

```diff
@@ -0,0 +1,200 @@
+
+    def run(
+        self,
+        model_id: str,
+        train_params: dict = {},
+        tags: dict = {},
+    ) -> pd.DataFrame:
+        """Run configured benchmark with given model and return the result.
+
+        Args:
+            model_id (str): Target model identifier
+            train_params (dict): Overrides global benchmark train params
+            tags (dict): Overrides global benchmark tags
+
+        Retruns:
+            pd.DataFrame: Table with benchmark metrics
+        """
+
+        # Build config file
+        cfg = self._build_config(model_id, train_params, tags)
+        cfg_dir = Path(cfg["output_path"])
+        cfg_dir.mkdir(parents=True, exist_ok=True)
+        cfg_path = cfg_dir / "cfg.yaml"
+        with open(cfg_path, "w") as cfg_file:
+            yaml.dump(cfg, cfg_file, indent=2)
+        cmd = [
+            "python",
+            "tools/experiment.py",
+            "-f",
+            cfg_path,
+        ]
+        if self.dry_run:
+            cmd.append("-d")
+        # Run benchmark
+        check_run(cmd)
+        # Load result
+        result = self.load_result(cfg_dir)
+        return result
+
+    @staticmethod
+    def load_result(result_path: str) -> pd.DataFrame | None:
+        """Load benchmark results recursively and merge as pd.DataFrame.
+
+        Args:
+            result_path (str): Result directory or speicific file.
+
+        Retruns:
+            pd.DataFrame: Table with benchmark metrics & options
+        """
+        # Search csv files
+        if os.path.isdir(result_path):
+            csv_file_paths = glob.glob(f"{result_path}/**/exp_summary.csv", recursive=True)
+        else:
+            csv_file_paths = [result_path]
+        results = []
+        # Load csv data
+        for csv_file_path in csv_file_paths:
+            result = pd.read_csv(csv_file_path)
+            # Append metadata if any
+            cfg_file_path = Path(csv_file_path).parent / "cfg.yaml"
+            if cfg_file_path.exists():
+                with cfg_file_path.open("r") as cfg_file:
+                    tags = yaml.safe_load(cfg_file).get("tags", {})
+                    for k, v in tags.items():
+                        result[k] = v
+            results.append(result)
+        if len(results) > 0:
+            return pd.concat(results, ignore_index=True)
+        else:
+            return None
+
+    def _build_config(
+        self,
+        model_id: str,
+        train_params: dict = {},
+        tags: dict = {},
+    ) -> dict:
+        """Build config for tools/expeirment.py."""
+        all_train_params = self.train_params.copy()
+        all_train_params.update(train_params)
+        all_tags = self.tags.copy()
+        all_tags.update(tags)
+
+        cfg = {}
+        cfg["tags"] = all_tags  # metadata
+        cfg["output_path"] = os.path.abspath(Path(self.output_root) / "-".join(list(all_tags.values()) + [model_id]))
+        cfg["constants"] = {
```

Comment:
For me, `constants['data_root']` seems not good. How about `data['data_root']` or `data['root']`?

Reply:
It's a syntax of yaml file for tools/experiment.py